sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -143,30 +138,25 @@ public bool TryGetProperty(string name, out object? value)
         /// <param name="value">The property value.</param>
         public void SetProperty(string name, object value)
         {
-            _typeProperties ??= new Dictionary<Type, object>();
-            Dictionary<string, object> properties;
-            if (!_typeProperties.TryGetValue(typeof(MessagePropertyKey), out var rawValue))
-            {
-                properties = new Dictionary<string, object>();
-                _typeProperties[typeof(MessagePropertyKey)] = properties;
-            }
-            else
-            {
-                properties = (Dictionary<string, object>)rawValue!;
-            }
-            properties[name] = value;
+            _properties ??= new Dictionary<string, object>();
+            _properties[name] = value;
```

Comment:
Why do we still use the dictionary here?

Reply:
Both APIs are public now so I don't really see the point of optimizing only one of them.

Reply:
Good point - we could use it for both.